src/System.Net.Security/src/System/Net/Security/Sslstream.Implementation.cs

```diff
@@ -159,256 +147,17 @@ internal void ValidateCreateContext(SslAuthenticationOptions sslAuthenticationOp
-        {
-            get
-            {
-                CheckThrow(true);
-                SslConnectionInfo info = Context.ConnectionInfo;
-                if (info == null)
-                {
-                    return (ExchangeAlgorithmType)0;
-                }
-
-                return (ExchangeAlgorithmType)info.KeyExchangeAlg;
-            }
-        }
-
-        internal int KeyExchangeStrength
-        {
-            get
-            {
-                CheckThrow(true);
-                SslConnectionInfo info = Context.ConnectionInfo;
-                if (info == null)
-                {
-                    return 0;
-                }
-
-                return info.KeyExchKeySize;
-            }
-        }
-
-        internal SslProtocols SslProtocol
-        {
-            get
-            {
-                CheckThrow(true);
-                SslConnectionInfo info = Context.ConnectionInfo;
-                if (info == null)
-                {
-                    return SslProtocols.None;
-                }
-
-                SslProtocols proto = (SslProtocols)info.Protocol;
-                SslProtocols ret = SslProtocols.None;
-
-#pragma warning disable 0618 // Ssl2, Ssl3 are deprecated.
-                // Restore client/server bits so the result maps exactly on published constants.
-                if ((proto & SslProtocols.Ssl2) != 0)
-                {
-                    ret |= SslProtocols.Ssl2;
-                }
-
-                if ((proto & SslProtocols.Ssl3) != 0)
-                {
-                    ret |= SslProtocols.Ssl3;
-                }
-#pragma warning restore
+        private bool RemoteCertRequired => _context == null || _context.RemoteCertRequired;
 
-                if ((proto & SslProtocols.Tls) != 0)
-                {
-                    ret |= SslProtocols.Tls;
-                }
+        private object SyncLock => _context;
 
-                if ((proto & SslProtocols.Tls11) != 0)
-                {
-                    ret |= SslProtocols.Tls11;
-                }
+        private X509Certificate InternalLocalCertificate => _context.IsServer ? _context.LocalServerCertificate : _context.LocalClientCertificate;
 
-                if ((proto & SslProtocols.Tls12) != 0)
-                {
-                    ret |= SslProtocols.Tls12;
-                }
+        private bool IsShutdown => _shutdown;
 
-                if ((proto & SslProtocols.Tls13) != 0)
-                {
-                    ret |= SslProtocols.Tls13;
-                }
-
-                return ret;
-            }
-        }
-
-        internal Stream InnerStream
-        {
-            get
-            {
-                return _innerStream;
-            }
-        }
-
-        internal SslStreamInternal SecureStream
-        {
-            get
-            {
-                CheckThrow(true);
-                return _secureStream;
-            }
-        }
+        private Stream InternalInnerStream => _innerStream;
```

Comment:
Do these private properties that just return a field need to be properties?  Why can't the call sites just access the field directly?

Reply:
That one can actually be removed sorry, that was a hang over from when I had only removed SslState and not SslSTreamInternal